autogen/coding/docker_commandline_code_executor.py

```diff
@@ -164,36 +193,37 @@ def execute_code_blocks(self, code_blocks: List[CodeBlock]) -> CommandLineCodeRe
         files = []
         last_exit_code = 0
         for code_block in code_blocks:
-            lang = code_block.language
-            code = silence_pip(code_block.code, lang)
-
-            try:
-                # Check if there is a filename comment
-                filename = _get_file_name_from_content(code, Path("/workspace"))
-            except ValueError:
-                return CommandLineCodeResult(exit_code=1, output="Filename is not in the workspace")
+            lang = code_block.language.lower()
+            if lang not in self.SUPPORTED_LANGUAGES:
+                outputs.append(f"Unsupported language {lang}")
+                last_exit_code = 1
+                break
 
-            if filename is None:
-                # create a file with an automatically generated name
-                code_hash = md5(code.encode()).hexdigest()
-                filename = f"tmp_code_{code_hash}.{'py' if lang.startswith('python') else lang}"
+            execute_code = self.execution_policies.get(lang, False)
+            code = silence_pip(code_block.code, lang)
 
+            # Check if there is a filename comment
+            filename = (
+                _get_file_name_from_content(code, self._work_dir) or f"tmp_code_{md5(code.encode()).hexdigest()}.{lang}"
```

Comment:
We should make `_get_file_name_from_content` a public function.

Reply:
This function also raises a value error if the file does not exist, we should wrap it with a try-catch.